include/pytorch/tokenizers/regex.h

```diff
@@ -38,11 +38,24 @@ class IRegex {
 };
 
 /**
- * @brief Creates a regex instance. Tries RE2 first, falls back to std::regex.
+ * @brief Creates a regex instance. If no strong symbol defined, only
+ * uses RE2. This is a weak symbol to allow other regex libraries to be
+ * used.
  *
  * @param pattern The regex pattern to compile.
  * @return A unique pointer to an IRegex-compatible object.
  */
 Result<std::unique_ptr<IRegex>> create_regex(const std::string& pattern);
```

Comment:
Why not just make this the weak symbol?

Reply:
Because we can't have RE2 in both weak implementation and strong implementation